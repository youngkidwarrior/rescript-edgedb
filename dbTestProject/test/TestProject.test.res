open TestFramework

let client = EdgeDB.Client.make()

external spawnSync: array<string> => 'a = "Bun.spawnSync"
external process: 'a = "process"

afterAllAsync(async () => {
  await client->EdgeDB.Client.close
})

describe("fetching data", () => {
  testAsync("fetching movies", async () => {
    expect(await client->Movies.allMovies)->Expect.toMatchSnapshot
  })

  testAsync("fetching single movie", async () => {
    expect(await client->Movies.movieByTitle(~title="The Great Adventure"))->Expect.toMatchSnapshot
  })

  testAsync("fetching non-existing movie", async () => {
    expect(
      await client->Movies.movieByTitle(~title="The Great Adventure 2"),
    )->Expect.toMatchSnapshot
  })
})

test("run unused selections CLI", () => {
  let res = spawnSync(["npx", "rescript-edgedb", "unused-selections"])
  expect(res["stdout"]["toString"]())->Expect.toMatchSnapshot
})
