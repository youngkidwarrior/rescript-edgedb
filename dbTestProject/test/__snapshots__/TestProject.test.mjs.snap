// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`fetching data fetching movies 1`] = `
[
  {
    "actors": [
      {
        "id": "a29aaa7e-5dd3-11ee-a767-3bd66df714fc",
        "name": "John Doe",
        "numberOfPets": 2,
      },
    ],
    "id": "a29acc34-5dd3-11ee-a767-fbf4b67abfe9",
    "title": "The Great Adventure",
  },
  {
    "actors": [
      {
        "id": "a29ab99c-5dd3-11ee-a767-e7331339dd97",
        "name": "Jane Smith",
        "numberOfPets": 1,
      },
    ],
    "id": "a29ad3aa-5dd3-11ee-a767-3f168b70341b",
    "title": "The Mystery",
  },
  {
    "actors": [
      {
        "id": "a29aaa7e-5dd3-11ee-a767-3bd66df714fc",
        "name": "John Doe",
        "numberOfPets": 2,
      },
      {
        "id": "a29ab99c-5dd3-11ee-a767-e7331339dd97",
        "name": "Jane Smith",
        "numberOfPets": 1,
      },
      {
        "id": "a29ac496-5dd3-11ee-a767-8b5ec9e4fb8d",
        "name": "Bob Johnson",
        "numberOfPets": 1,
      },
    ],
    "id": "a29ad954-5dd3-11ee-a767-7362fc90a02f",
    "title": "The Thriller",
  },
]
`;

exports[`fetching data fetching single movie 1`] = `
{
  "actors": [
    {
      "id": "a29aaa7e-5dd3-11ee-a767-3bd66df714fc",
      "name": "John Doe",
      "numberOfPets": 2,
    },
  ],
  "id": "a29acc34-5dd3-11ee-a767-fbf4b67abfe9",
  "title": "The Great Adventure",
}
`;

exports[`fetching data fetching non-existing movie 1`] = `undefined`;

exports[`run unused selections CLI 1`] = `
"Analyzing project... (this might take a while)

[31mâœ˜[0m Found 4 unused selections.

File "Movies":
   In query "AllMovies":
    - actors: numberOfPets
    - title
   In query "MovieByTitle":
    - actors: name
    - id
"
`;
